vimeo/vimeo-networking-java
# MPS-3052: Team permission PUT, DELETE methods.  PermissionPolicy GET methods

https://vimean.atlassian.net/browse/MPS-3052

# Summary
Added the following api methods, and any required Entities:

- fetchPermissionPolicyList
- fetchPermissionPolicy
- putTeamPermission
- deleteTeamPermission

## How to Test
Change the networking library version to 'MPS-3052-team-access-SNAPSHOT'.  You can use these code snippets to verify different endpoints work as expected:

```
//GET ALL PERMISSION POLICIES FOR THE TEAM
adaptRequest<PermissionPolicyList> {
    vimeoApiClient.fetchPermissionPolicyList(
        uri = "/users/79937352/permission_policies",
        fieldFilter = null,
        queryParams = null,
        cacheControl = null,
        callback = it
    )
}.subscribeBy {
    if (it is VimeoResponse.Success) {
        Log.i("PERMISSION POLICIES: " + it.data.toString())
    }
}.addTo(compositeDisposable)

//GET A SPECIFIC PERMISSION POLICY WITHIN THE TEAM
adaptRequest<PermissionPolicy> {
    vimeoApiClient.fetchPermissionPolicy(
        uri = "/users/79937352/permission_policies/58",
        fieldFilter = null,
        queryParams = null,
        cacheControl = null,
        callback = it
    )
}.subscribeBy {
    if (it is VimeoResponse.Success) {
        Log.i("PERMISSION POLICIES: " + it.data.toString())
    }
}.addTo(compositeDisposable)

//PUT A EDITOR PERMISSION FOR A TEAM MEMBER
adaptRequest<Unit> {
    vimeoApiClient.putTeamPermission(
        permissionPolicyUri = "/users/79937352/permission_policies/59",
        teamEntityType = TeamEntityType.TEAM_USER,
        teamEntityUri = "/users/79937352/team_users/770291",
        uri = "/users/79937352/projects/5826885/team_permission",
        callback = it,
        queryParams = null,
        bodyParams = null
    )
}.subscribeBy {
    if (it is VimeoResponse.Success) {
        Log.i("Folder Editor PERMISSION ADDED SUCCESSFULLY")
    }
}.addTo(compositeDisposable)

//PUT A VIEWER PERMISSION FOR A TEAM MEMBER
adaptRequest<Unit> {
    vimeoApiClient.putTeamPermission(
        permissionPolicyUri = "/users/79937352/permission_policies/60",
        teamEntityType = TeamEntityType.TEAM_USER,
        teamEntityUri = "/users/79937352/team_users/770291",
        uri = "/users/79937352/projects/5826885/team_permission",
        callback = it,
        queryParams = null,
        bodyParams = null
    )
}.subscribeBy {
    if (it is VimeoResponse.Success) {
        Log.i("Folder Viewer PERMISSION ADDED SUCCESSFULLY")
    }
}.addTo(compositeDisposable)

//REMOVE ANY TEAM PERMISSION FOR A TEAM MEMBER
adaptRequest<Unit> {
    vimeoApiClient.deleteTeamPermission(
        teamEntityType = TeamEntityType.TEAM_USER,
        teamEntityUri = "/users/79937352/team_users/770291",
        uri = "/users/79937352/projects/5826885/team_permission",
        callback = it,
        queryParams = null,
        bodyParams = null
    )
}.subscribeBy {
    if (it is VimeoResponse.Success) {
        Log.i("PERMISSION ADDED SUCCESSFULLY")
    }
}.addTo(compositeDisposable)
```

Endpoints from the prev PR if you want to check those our again(for regression testing):

```
//GET CURRENTLY ASSOCIATED TEAM PERMISSIONS FOR THE RESOURCE
adaptRequest<TeamPermissionList> {
    vimeoApiClient.fetchTeamPermissions(
        uri = "/users/79937352/projects/5826885/team_permissions",
        fieldFilter = null,
        queryParams = null,
        cacheControl = null,
        callback = it
    )
}.subscribeBy {
    if (it is VimeoResponse.Success) {
        Log.i("TEAM PERMISSIONS: " + it.data.toString())
    }
}.addTo(compositeDisposable)

//GET TEAM PERMISSIONS WITH A QUERY
adaptRequest<TeamPermissionList> {
    vimeoApiClient.fetchTeamPermissions(
        uri = "/users/79937352/projects/5826885/team_permissions",
        fieldFilter = null,
        queryParams = mapOf("query" to "ed"),
        cacheControl = null,
        callback = it
    )
}.subscribeBy {
    if (it is VimeoResponse.Success) {
        Log.i("TEAM PERMISSIONS: " + it.data.toString())
    }
}.addTo(compositeDisposable)
```
